spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+include SessionsHelper
+
+RSpec.describe PostsController, type: :controller do
+  before do
+    Rails.application.load_seed
+  end
+
+  let(:user) { User.find_by(username: 'user1') }
+
+  context 'GET #new' do
+    context 'when user is logged in but tries access new post page' do
+      it 'successfully shows the new post page' do
+        log_in(User.first)
+        get :new
+        expect(response).to have_http_status(:success)
+      end
+    end
+
+    context 'when user is NOT logged in but tries access new post page' do
+      it 'redirects to login page' do
+        # not logged in
+        get :new
+        expect(response).to redirect_to(login_url)
+      end
+    end
+
+  end
+
+  context 'POST #create' do
+
+    context 'when new post info is valid' do
+      it 'creates the new post' do
+        log_in(User.first)
+        expect{post :create, params: { post: { title: 'Major News!', body: 'You wont believe who just said blah blah', user_id: 1} }}.to change{Post.count}.by(1)
+        expect(response).to redirect_to(root_url)
```

Comment:
You should also check if new post is in DB

e.g.
to change by 1 like 
https://github.com/PolarisTLX/members-only/pull/1/files#diff-6273922273da89f91dddd7ca0111600aR59

or/and
```
expect(Post.last.title).to eq ("Major News!")
```